feat: Enhance parameter handling in ROS2

TannerGilbert opened the pull request:
This PR tackles task 1 and 2 described in #15.

## Task 1
For task 1 it is enough to simply build with the `--symlink-install` flag which creates symlinks for all non-compiled resources like Python files (Reference: https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Colcon-Tutorial.html#build-the-workspace). 

## Task 2

The parameter format expected for ROS changed between version 1 and 2 (https://docs.ros.org/en/foxy/How-To-Guides/Parameters-YAML-files-migration-guide.html). I don't think it makes sense to duplicate all config files though. Rather I opted to load the config file manually with the `yaml-cpp` library. The only thing I'm not that satisfied with is the parsing of the parameters to the desired type. Any suggestions in that regard are highly welcome.

Behaviour clarification:
Similar to the ROS 1 version the parameters from the YAML file have higher priority than the parameters specified in the launch file

Daehan2Lee commented:
Hi @TannerGilbert 
I tested the changes and everything works great — thank you so much for the excellent work!

I just have a couple of minor suggestions:

- Task 1: In the README.md, there's a small typo — `--system-install` should be corrected to `--symlink-install`.

- Task 2: When passing the `config_file` argument, the ROS1 version of genz-icp accepts just the file name (e.g., `kitti.yaml`), but in the current ROS2 implementation, an absolute path is required. It would be great if this could be adjusted to behave like ROS1.

Thanks again for your contribution!

TannerGilbert commented:
Thanks for the quick review. I fixed the typo in the README. Regarding the `config_file` path I added the option to specify just the name as you suggested to behave the same as ROS 2 but additionaly I made sure that you can still input an absolute path so that config files don't necessarily need to be in the `config` folder.

Daehan2Lee commented:
Awesome, everything is working perfectly now!
Thanks a lot for the quick fix — really appreciate it!!
